docs/specifications/tests/Basic-TP/basic01.md

```diff
@@ -5,48 +5,165 @@
+   2. The existance of the *child zone* has been determined. Emit the message
+      *[CHILD_FOUND]*.
+   3. Exit the steps.
+
+5. If the recursive lookup reaches a name server that authoritatively responds
+   (AA flag set) with NXDOMAIN for the child domain (*child zone*): 
+   1. The zone returning NXDOMAIN is considered to be the parent zone. Emit the
+      message *[PARENT_FOUND]*.
+   2. The non-existance of the *child zone* has been determined. 
+   3. If *normal test* emit the message *[NO_CHILD]*, if *undelegated test*
+      emit the message *[UNDEL_AND_NO_CHILD]*.
+   4. Exit the steps.
+
+6. If the recursive lookup reaches authorititative NOERROR answer (AA flag set), 
+   with no record in the answer section (NODATA):
+   1. The zone returning authoritative data is considered to be the parent zone. Emit the
+      message *[PARENT_FOUND]*.
+   2. The non-existance of the *child zone* has been determined.
+   3. If *normal test* emit the message *[NO_CHILD]*, if *undelegated test*
+      emit the message *[UNDEL_AND_NO_CHILD]*.
+   4. Exit the steps.
+
+7. If the recursive lookup reaches a non-authorititative NOERROR answer (AA flag 
+   unset), with a CNAME or DNAME record in the answer section:
+   1. A CNAME (DNAME) query with the RD flag unset is sent to the same server.
+   2. If the lookup returns an authoritative answer with a CNAME (DNAME) with
+      *child zone* name as owner name, then continue with steps 8, else continue
+      with next server in previous step.
+
+8. If the recursive lookup reaches an authorititative NOERROR answer (AA flag 
+   set), with a CNAME or DNAME record in the answer section:
+   1. The zone returning authoritative data is considered to be the parent zone. Emit the
+      message *[PARENT_FOUND]*.
+   2. The non-existance of the *child zone* has been determined.
+   3. If *normal test* emit the message *[NO_CHILD]*, if *undelegated test*
+      emit the message *[UNDEL_AND_NO_CHILD]*.
+   4. Exit the steps.
+
+9. If the recursive lookup reaches an authorititative NOERROR answer (AA flag 
+   set), with an SOA record with owner name child domain in the answer section:
+   1. The zone the previous delegation is considered to be the parent zone. Emit the
+      message *[PARENT_FOUND]*.
+   2. The existance of the *child zone* has been determined. Emit the message
+      *[CHILD_FOUND]*.
+   3. Exit the steps.
+
+10. If the server does not respond, the respond contain an unexpected RCODE or
+    any other error, try next server at step 2. 
+
+11. If delegation to a zone at a higher level is returned, then follow the
+    delegation.
+
+12. If all servers are exhausted: 
+    1. The parent zone cannot be determined.
+    2. The *child zone* cannot be determined.
+    3. If *normal test* emit the messages *[NO_CHILD]* and *[PARENT_INDETERMINED]*.
+    4. If *undelegated test* emit the messages *[UNDEL_AND_NO_CHILD]* and
+       *[UNDEL_AND_PARENT_INDETERMINED]*.
```

Comment:
Add the below line: 
4.a otherwise emit the messages *[NO_CHILD]* and *[PARENT_INDETERMINED]*.